sat/cli/bootprep/input/session_template.py

```diff
@@ -121,11 +121,16 @@ def validate_configuration_exists(self, **_):
             return
 
         try:
-            self.cfs_client.get_configuration(self.configuration)
+            resp = self.cfs_client.get('configurations', self.configuration, raise_not_ok=False)
+            if not resp.ok:
+                if resp.status_code == 404:
+                    raise InputItemValidateError(f'Configuration {self.configuration} specified for '
+                                                 f'{self} does not exist.')
+                else:
+                    self.cfs_client.raise_from_response(resp)
         except APIError as err:
-            # TODO: should probably differentiate between 404 Not Found and other errors
```

Comment:
There are still two more instances of this TODO comment in this file on lines 233 and 308 where we're checking if the image specified in a session template exists.